x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterService.java

```diff
@@ -406,7 +411,9 @@ public void onFailure(Exception e) {
 
                 @Override
                 public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
-                    listener.onResponse(TrainedModelAssignmentMetadata.fromState(newState));
+                    if (isUpdated) {
+                        listener.onResponse(TrainedModelAssignmentMetadata.fromState(newState));
+                    }
```

Comment:
Ah, yeah, this would happen because of the `return currentState` in execute kicking off another rebalancing.

Reply:
Yeah, you were right there was something off there!